app/models/group_order_invoice.rb

```diff
@@ -0,0 +1,60 @@
+class GroupOrderInvoice < ApplicationRecord
+
+  belongs_to :group_order
+  validates_presence_of :group_order
+  validates_uniqueness_of :invoice_number
+  validate :tax_number_set
+  after_initialize :init, unless: :persisted?
+
+  def generate_invoice_number(count)
+    trailing_number = count.to_s.rjust(4, '0')
+    unless  GroupOrderInvoice.find_by(invoice_number: Time.now.strftime("%Y%m%d") + trailing_number)
+      Time.now.strftime("%Y%m%d") + trailing_number
+    else
+      generate_invoice_number(count.to_i + 1)
+    end
+  end
+
+  def tax_number_set
+    if FoodsoftConfig[:contact][:tax_number].blank?
+      errors.add(:group_order_invoice, "Keine Steuernummer in FoodsoftConfig :contact gesetzt")
+    end
+  end
+
+  def init
+    self.invoice_number = generate_invoice_number(1) unless self.invoice_number
+    self.invoice_date = Time.now unless self.invoice_date
+    self.payment_method = FoodsoftConfig[:group_order_invoices]&.[](:payment_method) || I18n.t('activerecord.attributes.group_order_invoice.payment_method') unless self.payment_method
+  end
+
+  def name
+    I18n.t('activerecord.attributes.group_order_invoice.name') + "_#{invoice_number}"
```

Comment:
the translation should be moved from the model to a controller, since the model should be language independent